app/db/impl/access/UserBase.scala

```diff
@@ -153,21 +155,20 @@ class UserBase(override val service: ModelService,
     }
   }
 
-
   /**
-    * Returns the currently authenticated User.
+    * Returns the currently authenticated User.c
     *
     * @param session  Current session
     * @return         Authenticated user, if any, None otherwise
     */
   def current(implicit session: Request[_], ec: ExecutionContext): Future[Option[User]] = {
     session.cookies.get("_oretoken") match {
-    case None => Future(None)
-    case Some(cookie) => getSession(cookie.value).flatMap {
-        case None => Future(None)
-        case Some(s) => s.user
+      case None => Future.successful(None)
```

Comment:
flatMap

Reply:
how? thats option future option

Reply:
Ah, ok. In that case let's wait til OptionT, or you can use a fold (`session.cookies.get("_oretoken").fold[Future[Option[User]](Future.successful(None))(cookie => getSession(cookie.value).flatMap(_.user))`)